### PR TITLE
AP_GPS: Force the baudrate on SBF

### PR DESCRIPTION
**Pull request by WickedShell**
This rewrites the SBF config loop to force the baudrate on the GPS rather then trusting whatever it was detected at (which prevents us from sticking with a GPS at 9600 baud, which is guaranteed to not work correctly). It also pushes the typical SBF unit up from 115200 to 230400, which has the general benefit of reducing lag on the data, and allowing us to get RTCM corrections to the unit faster (if doing moving baseline this is essential).

Other incidental notes with this:
 - whitespace trimming the config strings saves about 16 bytes
 - the config loop now allocates and add's the `\n` character for each string as needed, this both saves some space, but more importantly actually significantly simplified the deallocation of those strings we have to make on the fly

This was tested upon a HitecMosaic.

### CODE REVIEW

**Comment by rrr6399** [comment]
Did you all discover recently that 460,800 wasn't required for the RTK moving baseline mode?

**Comment by WickedShell** [comment]
> Did you all discover recently that 460,800 wasn't required for the RTK moving baseline mode?

Honestly I just never pushed it any higher when I did all the SBF for it at the end of last year, as I didn't need to. (I started at 115200, and did have problems there though). I can't really comment on anyone else's recent work with it, but since most SBF users have been using 115200 as that's just the default it was worth calling out here.

**Comment by rrr6399** [comment]
OK. U-blox had recommended 460,800 for the F9P moving base, but I noticed that the baud rate was recently lowered to 230,400 to address some DMA issue.

File: libraries/AP_HAL_ChibiOS/hwdef/HitecMosaic/hwdef.dat [hunk 1]
```diff
@@ -172,5 +172,5 @@ define HAL_PERIPH_ENABLE_BARO
 define HAL_GPS_TYPE_DEFAULT 10
 define HAL_GPS_COM_PORT_DEFAULT 3
 define GPS_SBF_STREAM_NUMBER 3
-define GPS_SBF_EXTRA_CONFIG "sdfa, DSK1, DeleteOldest\n", "sfn, DSK1, Incremental, Mosaic\n", "sfno, off\n"
+define GPS_SBF_EXTRA_CONFIG "sdfa, DSK1, DeleteOldest", "sfn, DSK1, Incremental, Mosaic\n", "sfno, off"
```
Comment (khancyr) [inline libraries/AP_HAL_ChibiOS/hwdef/HitecMosaic/hwdef.dat:175]: shouldn't the \n after Mosaic be removed too ?

Comment (WickedShell) [reply libraries/AP_HAL_ChibiOS/hwdef/HitecMosaic/hwdef.dat]: Yeah, I missed it with the multiple items to a line here. It doesn't actually change the build size due to alignment reasons, but it's still better to remove it here. Fixed it up.

**Review by tridge — APPROVED** [review]

**Pull request merged**